src/serializer/ResidualHeapVisitor.js

```diff
@@ -188,6 +188,7 @@ export class ResidualHeapVisitor {
 
   visitObjectProperties(obj: ObjectValue, kind?: ObjectKind): void {
     let { skipPrototype, constructor } = getObjectPrototypeMetadata(this.realm, obj);
+    if (obj.temporalAlias !== undefined) return;
```

Comment:
The serializer will process the `.temporalAlias`: `this.serializeValue(val.temporalAlias);`. However, the `.temporalAlias` doesn't get visited in the ResidualHeapVisitor. Maybe it does get visited as a side effect of something, but even then, it should get actively visited to get put into the right scope.

Reply:
The temporal alias should be visited from the generator that was active at the temporal point where it was created.